activerecord/test/cases/tasks/mysql_rake_test.rb

```diff
@@ -340,6 +359,26 @@ def test_structure_load
 
         ActiveRecord::Tasks::DatabaseTasks.structure_load(@configuration, filename)
       end
+
+      def test_structure_load
+        filename = "awesome-file.sql"
+        Kernel.expects(:system).with("mysql", "--execute", %{SET FOREIGN_KEY_CHECKS = 0; SOURCE #{filename}; SET FOREIGN_KEY_CHECKS = 1}, "--database", "test-db", "--noop")
+          .returns(true)
```

Comment:
We've been trying to move away from Mocha (admittedly stalled at the moment) and have introduced these assertions to help with that: https://github.com/rails/rails/blob/2fd3034bf1862bb8e40c0a0a162316b8ff564d0e/activesupport/lib/active_support/testing/method_call_assertions.rb

Can we switch to those in these new tests?